src/shared/util/validation.ts

```diff
@@ -7,6 +7,10 @@ export const WHITELIST = [new RegExp('^http://localhost:4566')]
 
 export const SHORT_URL_REGEX = /^[a-z0-9-]+$/
 
+export const TAG_STRING_REGEX = /^[A-Za-z0-9-]+$/
```

Comment:
we need to support underscore as well

Reply:
checking our [notion doc](https://www.notion.so/opengov/As-a-user-I-want-to-add-tags-while-creating-link-s-so-that-I-can-identify-the-links-easily-7d9b280c038043b9b2210b8eba045fff), it looks like we're not looking to support special characters apart from hyphens! I also modelled this validation on the corresponding section in [this PR](https://github.com/opengovsg/GoGovSG/pull/1917/files#diff-691549b01fd4fb725dccc9297b71b9bdbcee326f2fbe70acc015c235d42a63a8)

Reply:
(update: we're putting underscores in - taking in the new tag regexes after merging in new changes from `develop`!)